### DISCUSSION TITLE
Use Primitive as connection in Custom Node

### DISCUSSION THREAD

**Discussion by rhalaly** [comfyanonymous/ComfyUI#4619] [Q&A]
I want to create a custom node that needs to get a string, but as connection. When I use

```
"required": {
                "text": ("STRING", ),
            }
```

I get the `text` as textbox widget. How can I get that field as connection?

**Comment by christian-byrne** [comment]
`forceInput` field:

```python
    "required": {
        "text": ("STRING", ),
        "forceInput": True
    }
```

**Reply by ltdrdata** [reply]
or you can use "defaultInput".
They are slightly different. forceInput doesn't allow convert back to widget.
